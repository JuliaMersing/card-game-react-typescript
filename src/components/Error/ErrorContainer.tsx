import React, { useState } from 'react';

interface ErrorProps {
	message: string;
	onClose?: () => void; // Add the onClose prop to the interface
}

export const ErrorContainer: React.FunctionComponent<ErrorProps> = ({
	message,
	onClose,
}: ErrorProps) => {
	const [showAlert, setShowAlert] = useState(true);

	return (
		<>
			{showAlert ? (
				<div className="container-error">
					<span className="error-message">{message}</span>
					<button
						className="button-error"
						onClick={() => {
							setShowAlert(false);
							if (onClose) {
								onClose();
							}
						}}
					>
						<span>x</span>
					</button>
				</div>
			) : null}
		</>
	);
};

export const ClosingErrorContainer: React.FunctionComponent = () => {
	return (
		<>
			<ErrorContainer message="You have to draw a card first!" />
		</>
	);
};
