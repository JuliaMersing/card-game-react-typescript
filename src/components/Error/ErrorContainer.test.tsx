import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ErrorContainer } from './ErrorContainer';

describe('ErrorContainer component', () => {
	test('should match snapshot', () => {
		render(<ErrorContainer message="You have to draw a card first!" />);
		const error = screen.getByText('You have to draw a card first!');
		expect(error).toMatchSnapshot();
	});

	test('should call onClose when button is clicked', () => {
		const mockHandleOnClose = jest.fn();
		render(
			<ErrorContainer
				message="You have to draw a card first!"
				onClose={mockHandleOnClose}
			/>
		);

		const closeButton = screen.getByRole('button');
		userEvent.click(closeButton);

		expect(mockHandleOnClose).toHaveBeenCalledTimes(1);
	});
});
