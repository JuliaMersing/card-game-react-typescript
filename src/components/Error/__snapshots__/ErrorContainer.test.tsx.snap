// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorContainer component should match snapshot 1`] = `
<span
  class="error-message"
>
  You have to draw a card first!
</span>
`;
